### ISSUE TITLE
`serverMiddleware` stopped being triggered in Nuxt3 RC3 & RC4

### ISSUE THREAD

**Issue by emrekaraca** [nuxt/framework#5756]
### Environment

```
❯ npx nuxi info
Nuxt CLI v3.0.0-rc.4                                                    17:43:13
RootDir: /home/projects/nuxt-starter-zq6e6c                             17:43:22
Nuxt project info:                                                      17:43:22

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.4`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `serverMiddleware`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------
```

### Reproduction

Working reproduction (Nuxt3 RC2): https://stackblitz.com/edit/nuxt-starter-wvdsp7
Broken reproduction (Nuxt3 RC4): https://stackblitz.com/edit/nuxt-starter-zq6e6c

||Working reproduction (Nuxt3 RC2)|Broken reproduction (Nuxt3 RC4)|
|---|---|---|
|Link|https://stackblitz.com/edit/nuxt-starter-wvdsp7|https://stackblitz.com/edit/nuxt-starter-zq6e6c|
|Screenshot|![Screenshot 2022-07-06 at 17 45 19](https://user-images.githubusercontent.com/19314055/177590852-c9ecd6d7-7679-4f42-a89e-40ca29ec9c35.png)|![Screenshot 2022-07-06 at 17 45 09](https://user-images.githubusercontent.com/19314055/177590814-e5b45570-ba51-41aa-8d53-c2926e7fb12f.png)|

### Describe the bug

In Nuxt3 RC3 & RC4 the `serverMiddleware` defined via `nuxt.config.ts` stopped being triggered for the specified paths. Instead of executing the middleware, Nuxt responds with a rendered page.

This works in RC1 & RC2.

### Additional context

In my Nuxt3 app, I register an `express` instance as `serverMiddleware` via the field in `nuxt.config.ts`. Until RC2, the specified paths trigger the api routes defined via `express`. From RC3, the `express` instance is not reached anymore and the default catch all response is returned instead (HTML document)

**Comment by danielroe** [comment]
Yes, there was a change upstream in nitro. Try this: https://stackblitz.com/edit/nuxt-starter-fjhtpr.

**Comment by emrekaraca** [comment]
Thanks a lot @danielroe!
